engine/src/main/java/org/terasology/input/cameraTarget/CameraTargetSystem.java

```diff
@@ -144,7 +144,7 @@ private void updateFocalDistance(HitResult hitInfo, float delta) {
         if (hitInfo.isHit()) {
             Vector3f playerToTargetRay = new Vector3f();
             //calculate the distance from the player to the hit point
-            playerToTargetRay.sub(JomlUtil.from(hitInfo.getHitPoint()), localPlayer.getViewPosition());
+            hitInfo.getHitPoint().sub(JomlUtil.from(localPlayer.getViewPosition()), playerToTargetRay);
```

Comment:
I guess/hope that "rotation" is on purpose?
So TeraMath's `x.sub(y, z)` equals joml's `y.sub(z, x)`?

Reply:
for termath x.sub(y,z) --> x = y - z where joml it's  x-y = z. z is the dest in this case.